external/boost/archive/portable_binary_oarchive.hpp

```diff
@@ -52,6 +52,7 @@ class portable_binary_oarchive_exception :
         switch(code){
         case invalid_flags:
             msg = "cannot be both big and little endian";
+            /* FALLTHRU */

```

Comment:
I created this file (in PR #1531) by copying the code from [the Boost example directory](http://www.boost.org/doc/libs/1_64_0/libs/serialization/example/) which is unofficial and not sufficiently reviewed/tested. It seems that this class `portable_binary_oarchive_exception ` is poorly written compared to its counterpart in [`portable_binary_iarchive.hpp`](https://github.com/monero-project/monero/blob/4466b6d1b08d827f3bdddb1cab8ca05833b65437/external/boost/archive/portable_binary_iarchive.hpp#L41), and the proper definition would have been as follows:

```c++
class portable_binary_oarchive_exception : 
    public boost::archive::archive_exception
{
public:
    enum exception_code {
        invalid_flags 
    } m_exception_code;
    portable_binary_oarchive_exception(exception_code c = invalid_flags ) :
        boost::archive::archive_exception(boost::archive::archive_exception::other_exception),
        m_exception_code(c)
    {}
    virtual const char *what( ) const throw( )
    {
        const char *msg = "programmer error";
        switch(m_exception_code){
        case invalid_flags:
            msg = "cannot be both big and little endian";
            break;
        default:
            msg = boost::archive::archive_exception::what();
            assert(false);
            break;
        }
        return msg;
    }
};
```

@hyc I'm sorry to have caused this, but could you please add the above code to this PR?